### PR TITLE
ddl: The default value for null flag is yes when the column type is timestamp 

### PR DESCRIPTION
**Pull request by zimulala**
Fix issue #2606 
It needs to be merged with [PR](https://github.com/pingcap/tidb-test/pull/274) at the same time.

### CODE REVIEW

File: ddl/ddl_db_test.go [hunk 1]
```diff
@@ -833,6 +841,17 @@ func match(c *C, row []interface{}, expected ...interface{}) {
...
 
+func (s *testDBSuite) TestTimestamp(c *C) {
+	defer testleak.AfterTest(c)
+	s.tk = testkit.NewTestKit(c, s.store)
+	s.tk.MustExec("use " + s.schemaName)
+
+	s.tk.MustExec("create table test_timestamp (a int, b timestamp default null on update current_timestamp);")
+	s.mustExec(c, "insert into test_timestamp set a = 1")
+	rowStr := fmt.Sprintf("%v", nil)
+	s.tk.MustQuery("select b from test_timestamp").Check(testkit.Rows(rowStr))
```
Comment (coocood) [inline ddl/ddl_db_test.go]: update the row and check that it has been updated to current time.

Comment (zimulala) [inline ddl/ddl_db_test.go]: You said the test has been added in tidb-test. This PR handles the null flag, temporarily can not add this test, and I will add it at the next PR.

**Comment by hanfei1991** [comment]
LGTM

**Comment by coocood** [comment]
LGTM

**Comment by zimulala** [comment]
The default value of `null` flag is different according to the value of `explicit_defaults_for_timestamp`.
The default value for `explicit_defaults_for_timestamp` is 0 in MySQL 5.7.15 and MySQL 5.7.17, and 1 for MySQL 5.6.34 and MySQL 5.7.9. And this value can't be modified in MySQL 5.6.34.
The original code in line with the `explicit_defaults_for_timestamp` value of 0.
In view of above, we will not make any changes.

**Pull request closed without merging**